sdk/scheduler/src/test/java/com/mesosphere/sdk/config/ConfigurationUpdaterTest.java

```diff
@@ -304,9 +314,37 @@ public void testPodsUsersUnsetInPreviousServiceSpecButSetToRootInNewServiceSpec(
                 .pods(podsWithUsers)
                 .build();
 
-        ConfigurationUpdater.UpdateResult result =
-                configurationUpdater.updateConfiguration(SERVICE_SPECIFICATION_WITH_USER);
+        ConfigurationUpdater.UpdateResult result = null;
+        try {
+            result = configurationUpdater.updateConfiguration(SERVICE_SPECIFICATION_WITH_USER);
+        } catch (ConfigStoreException e) {
+            Assert.fail("Expected ConfigStoreException not be thrown here.");
+            Assert.assertFalse(e.getMessage().contains("Cannot change existing pod type user"));
+        }
+        Assert.assertNotNull(result);
         Assert.assertEquals(TARGET_ID, result.getTargetId());
         Assert.assertEquals(0, result.getErrors().size());
     }
+
+    @Test
+    public void testServiceUserChangedInNewServiceSpec() throws ConfigStoreException {
+        final ConfigurationUpdater<ServiceSpec> configurationUpdater = new DefaultConfigurationUpdater(
+                mockStateStore,
+                mockConfigStore,
+                DefaultServiceSpec.getComparatorInstance(),
+                DefaultConfigValidators.getValidators(SchedulerConfigTestUtils.getTestSchedulerConfig()),
+                Optional.empty());
+        when(mockConfigStore.getTargetConfig()).thenReturn(TARGET_ID);
+
+        when(mockConfigStore.fetch(TARGET_ID)).thenReturn(ORIGINAL_SERVICE_SPECIFICATION_WITH_USER);
+
+        //ConfigurationUpdater.UpdateResult result = null;
```

Comment:
this unused code can be removed ?! (Same in other places)

Reply:
whoops! fixed now, thanks.